clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -172,6 +172,27 @@ Decl *SemaHLSL::ActOnStartBuffer(Scope *BufferScope, bool CBuffer,
   return Result;
 }
 
+static unsigned calculateLegacyCbufferFieldAlign(const ASTContext &Context,
+                                                 QualType T) {
+  // Aggregate types are always aligned to new buffer rows
+  if (T->isAggregateType())
+    return 16;
+
+  assert(Context.getTypeSize(T) <= 64 &&
```

Comment:
After determining this is not an aggregate, we still need to get the scalar element before proceeding if this is a vector, right?

Reply:
Ah it seems I have incorrectly assumed that vector is considered an aggregate, I need to go handle that case

Reply:
I suppose even if that assumption is/where true it would still do the wrong layout